framework/wazuh/configuration.py

```diff
@@ -723,3 +704,62 @@ def upload_group_file(group_id, tmp_file, file_name='agent.conf'):
         return upload_group_configuration(group_id, file_data)
     else:
         raise WazuhException(1111)
+
+
+def get_active_configuration(agent_id, component, configuration):
+    """
+    Reads agent loaded configuration in memory
+    """
+    if not component or not configuration:
+        raise WazuhException(1307)
+
+    sockets_path = common.ossec_path + "/queue/ossec/"
+
+    components = ["agent", "agentless", "analysis", "auth", "com", "csyslog", "integrator", "logcollector", "mail",
+                  "monitor", "request", "syscheck", "wmodules"]
+
+    # checks if the component is correct
+    if component not in components:
+        raise WazuhException(1101, "Invalid target")
+
+    if component == "analysis" and (configuration == "rules" or configuration == "decoders"):
+        raise WazuhException(1101, "Could not get requested section")
+
+    if agent_id == '000':
+        dest_socket = sockets_path + component
+        command = "getconfig " + configuration
+    else:
+        dest_socket = sockets_path + "request"
+        command = str(agent_id).zfill(3) + " " + component + " getconfig " + configuration
+
+    # Socket connection
+    try:
+        s = OssecSocket(dest_socket)
+    except Exception as e:
+        if agent_id == '000':
+            raise WazuhException(1013, "The component might be disabled")
+        else:
+            raise WazuhException(1013, str(e))
+
+    # Send message
+    s.send(command.encode())
+
+    # Receive response
+    try:
+        # Receive data length
+        data = s.receive().decode().split(" ", 1)
```

Comment:
These lines are not covered by unit tests.

Reply:
Done in 37432d86ad22e5a26ea27e7964a2b5523eb379e7